db/migrate/20230803225825_container_indexes.rb

```diff
@@ -0,0 +1,25 @@
+class ContainerIndexes < ActiveRecord::Migration[6.1]
+  def change
+    # container projects page
+    add_index :container_routes, :container_project_id
+    add_index :container_services, :container_project_id
+    # didn't show much, since only had 1 record
+    add_index :container_replicators, :container_project_id
+
+    add_index :container_groups, [:container_project_id, :id], :name => "index_container_groups_on_cpid_id_not_del", :where => "deleted_on IS NULL"
+    # container providers screen
+
+    # NOTE: this gets index only scans, but has a larger index size
+    add_index :taggings, [:taggable_type, :taggable_id, :tag_id], :name => "index_taggings_on_type_id_id"
+    remove_index :taggings, :column => [:taggable_id, :taggable_type], :name => "index_taggings_on_taggable_id_and_taggable_type"
```

Comment:
I don't understand this at all.  Typically pattern for indexes is to put the highest cardinality first, unless there are queries where the lower cardinality one can stand alone.  Also I don't understand adding tag_id

Reply:
I'm not against adding the tag_id - I'm questiong why taggable_type and taggable_id order were reversed - that's the part I'm questioning.